### DISCUSSION TITLE
fruit sequence for catch goes wrong after spinner (banana) in lazer

### DISCUSSION THREAD

**Discussion by Nekomosh004** [ppy/osu#20684] [Ideas] [answered]
This is something I noticed everytime I watch replays for both stable and lazer. After a few minutes of gameplay, I noticed the fruit no longer match. I checked why this is happening, and it seems like lazer breaks the pattern when a spinner (banana) was introduced. 

The fruit sequence for catch in stable are as follows:
Green Pear → Purple Grape → Red Apple → Orange

This pattern does not change even with a new combo. However for some reason, the pattern breaks when a spinner (banana) was introduced. Lazer skips one fruit sequence after a Banana.

In stable this what happens:
Purple Grape → Red Apple → Orange → **Green Pear** → **Banana** → **Purple Grape** → Red Apple → Orange → Green Pear (and so on)

However, lazer does this:
Purple Grape → Red Apple → Orange → **Green Pear** → **Banana** → **Red Apple** (skipped Purple Grape) → Orange → Green Pear (and so on)

I believe this can be reproduced on any beatmap playable for catch (convert or catch specific beatmap), but I specifically noticed this on [this beatmap](https://osu.ppy.sh/beatmapsets/716213#fruits/1513229) during the 198th combo

Here's a scuffed video edit of what it looks like:

https://user-images.githubusercontent.com/82471957/194832499-e4e211c6-b219-41ef-9b34-a64aa07ed694.mp4

I hope this could be fixed. Thank you! c:

**Comment by ghost** [comment, marked as answer]
#19998 might be related

**Reply by peppy** [reply]
Definitely the cause.
